nn/math_.py

```diff
@@ -195,7 +195,7 @@ def gating(x: nn.Tensor, *, axis: Optional[nn.Dim] = None,
 
 # noinspection PyShadowingBuiltins,PyShadowingNames
 def combine(
-             *sources: Union[nn.Tensor, nn.RawTensorTypes],
+             *sources: Union[Sequence[nn.Tensor], Union[nn.RawTensorTypes]],
```

Comment:
First of all, I assume you mean this?
```suggestion
             *sources: Sequence[Union[nn.Tensor, nn.RawTensorTypes]],
```
But then, I think this is not how it works. Did you actually check (e.g. in PyCharm)? As far as I know, and PyCharm also handles it this way, the type refers to each element of `sources` but not `sources` itself. So it is already correct.

Reply:
Yes I did. Without the change Pycharm marked `nn.combine([x, mu], kind="sub", allow_broadcast_all_sources=True)` where `x` and `mu` are tensors as wrong type.

```Expected type 'Union[Tensor, int, float, complex, number, ndarray, bool, str]', got 'List[Tensor]' instead```

Reply:
But `nn.combine([x, mu], kind="sub")` is wrong, and PyCharm gives the correct warning?

It is `nn.combine(x, mu, kind="sub")`.

With your change here, you would allow sth like `nn.combine([a, b], [c, d], kind="sub")` which does not really make sense.

Reply:
You are right. I did not see the change made to the input to the layer compared to `_combine`, where it worked like this, was assuming it stayed the same for now.